huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go

```diff
@@ -564,9 +759,59 @@ func resourceGatewayRead(_ context.Context, d *schema.ResourceData, meta interfa
 		d.Set("access_private_ip_2", utils.PathSearch("vpn_gateway.access_private_ip_2", getGatewayRespBody, nil)),
 	)
 
+	certificateContent := d.Get("certificate").([]interface{})
+	if len(certificateContent) == 1 {
+		certificateMap := certificateContent[0].(map[string]interface{})
+		certificateID := certificateMap["certificate_id"].(string)
+		getGatewayCertificatePath := getGatewayClient.Endpoint + getGatewayCertificateHttpUrl
+		getGatewayCertificatePath = strings.ReplaceAll(getGatewayCertificatePath, "{project_id}", getGatewayClient.ProjectID)
+		getGatewayCertificatePath = strings.ReplaceAll(getGatewayCertificatePath, "{gateway_id}", d.Id())
+		getGatewayCertificatePath = strings.ReplaceAll(getGatewayCertificatePath, "{certificate_id}", certificateID)
+		getGatewayCertificateOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+		}
+		getGatewayCertificateResp, err := getGatewayClient.Request("GET", getGatewayCertificatePath, &getGatewayCertificateOpt)
+		if err != nil {
+			return common.CheckDeletedDiag(d, err, "error retrieving Gateway certificate")
+		}
+		getGatewayCertificateRespBody, err := utils.FlattenResponse(getGatewayCertificateResp)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		mErr = multierror.Append(mErr,
+			d.Set("certificate", flattenGatewayCertificateResponse(d, getGatewayCertificateRespBody)),
+		)
+	}
 	return diag.FromErr(mErr.ErrorOrNil())
 }
 
+func flattenGatewayCertificateResponse(d *schema.ResourceData, resp interface{}) []interface{} {
+	rst := d.Get("certificate").([]interface{})
+	curJson, err := jmespath.Search("certificate", resp)
+	if err != nil {
+		log.Printf("[ERROR] error parsing certificate from response= %#v", resp)
+		return rst
+	}
+
+	rstCertificate := rst[0].(map[string]interface{})
+	rstCertificate["certificate_id"] = utils.PathSearch("id", curJson, nil)
+	rstCertificate["status"] = utils.PathSearch("status", curJson, nil)
+	rstCertificate["issuer"] = utils.PathSearch("issuer", curJson, nil)
```

Comment:
this is more golang style
```suggestion
	rstCertificate := map[string]interface{}{
		"certificate_id": utils.PathSearch("id", curJson, nil),
		"status":         utils.PathSearch("status", curJson, nil),
		"issuer":         utils.PathSearch("issuer", curJson, nil),
		...
	}
```

Reply:
must add in the schema map because the GET response missing the CREATE request body parameter.